Add dhclient basic configuration

avlasov-mos-de opened the pull request:
Allows configuring general section as well as configuring
    each dhcp enabled interface separetly.
    Does not allow alias or lease configuration.

cloudlab-autobuild commented:
Can one of the admins verify this patch?

fpytloun commented on linux/network/dhclient.sls:
```diff
@@ -0,0 +1,14 @@
+{%- from "linux/map.jinja" import network with context %}
+{%- if network.enabled %}
+
+{%- set dhclient = network.get('dhclient_config', False) %}
```
Name the key only `dhclient` to match your newly introduced role

fpytloun commented on linux/map.jinja:
```diff
@@ -110,6 +110,7 @@
            'host': 'none',
         },
         'host': {},
+        'dhclient_cfg_path': '/etc/dhcp/dhclient.conf',
```
I think dhclient_config would be sufficient but not mandatory.

fpytloun commented on README.rst:
```diff
@@ -772,6 +772,73 @@ OpenVswitch Bridges
             use_interfaces:
             - eth1
 
+DHCP client configuration
```
Also add new tests into `test/pillar` and check `make test`

fpytloun commented on linux/files/dhclient.conf:
```diff
@@ -0,0 +1,91 @@
+{# Macro definition #}
+{%- macro render_key(section, key) -%}
+{%- if section[key] is defined and section[key]|length > 0 %}
```
don't know if `section[key] is defined` is working as you expect, I would instead use `section.get(key)` that defaults to None.

avlasov-mos-de commented:
Thanks for feedback, I did  a couple of changes accordingly to comments. Also I rewrote macro a bit in order to support nested lists for some options. As for tests, I updated pillar/network.sls as it looks like this change has to be tested as part of network. make test finished with "Execution successful".  If I got it wrong could you explain what kind of tests I should add
